### PR TITLE
M0228J-41 Start MicroJS tests documentation.

### PR DESCRIPTION
**Pull request by bguedas**
Start MicroJS test documentation

### CODE REVIEW

File: ApplicationDeveloperGuide/microjs/testing.rst [hunk 1]
```diff
@@ -9,8 +9,46 @@
...
\ No newline at end of file
+A test is a JavaScript file with the ``.js`` extension. It must be put in  ``src/test/js`` directory of the project.
+
+Each JavaScript test file is considered as a single test.
+
+A test will pass if it can be successfully transpiled to Java and no exception is thrown during the test execution.
+
+Successfully transpiled tests are put in ``src-adpgenerated/microjs/java`` under the ``com.microej.microjstest`` package. Each generated test file is named with following pattern: ``_JsTest_<jsfilename>Code.java`` where ``<jsfilename>`` is the name of the JavaScript test file without its extension.
+
+A test that could not be successfully transpiled generates a test error entry in the class ``_JsTest__brokenCode.java``.
```
Comment (plaupretre) [inline ApplicationDeveloperGuide/microjs/testing.rst:23]: Maybe add "This broken test will then fail when running the tests"

File: ApplicationDeveloperGuide/microjs/testing.rst [hunk 2]
```diff
@@ -9,8 +9,46 @@
...
+  function assert(result) {
+    if (result == 0) {
+	  throw 'Test failed!'
+	}
+  }
+
+Running tests
+-------------
+
+Tests are run on simulator.
```
Comment (plaupretre) [inline ApplicationDeveloperGuide/microjs/testing.rst:43]: on the MicroEJ Simulator ?

File: ApplicationDeveloperGuide/microjs/testing.rst [hunk 3]
```diff
@@ -9,8 +9,46 @@
...
+
+All tests can be automatically executed when building the module. To do this, add the following lines to your ``module.ivy`` file between the ``<info>`` tags
```
Comment (plaupretre) [inline ApplicationDeveloperGuide/microjs/testing.rst:45]: inside the `<info>` tag instead of between the `<info>` tags ?

**Comment by bguedas** [comment]
This part is obsolete.

**Pull request closed without merging**